posthog/clickhouse/migrations/0111_add_soft_delete_column_on_groups.py

```diff
@@ -0,0 +1,32 @@
+from infi.clickhouse_orm import migrations
+
+from posthog.clickhouse.client.connection import get_client_from_pool
+from posthog.settings import CLICKHOUSE_CLUSTER
+
+
+DROP_COLUMNS_GROUPS = """
+ALTER TABLE {table} ON CLUSTER {cluster}
+DROP COLUMN IF EXISTS is_deleted
+"""
+
+ADD_COLUMNS_GROUPS = """
+ALTER TABLE {table} ON CLUSTER {cluster}
+ADD COLUMN IF NOT EXISTS is_deleted Boolean
+"""
+
+ADD_COLUMNS_INDEX_GROUPS = """
+ALTER TABLE {table} ON CLUSTER {cluster}
+ADD INDEX IF NOT EXISTS is_deleted_idx (is_deleted) TYPE minmax GRANULARITY 1
+"""
+
+
+def add_columns_to_required_tables(_):
+    with get_client_from_pool() as client:
+        client.execute(DROP_COLUMNS_GROUPS.format(table="groups", cluster=CLICKHOUSE_CLUSTER))
+        client.execute(ADD_COLUMNS_GROUPS.format(table="groups", cluster=CLICKHOUSE_CLUSTER))
+        client.execute(ADD_COLUMNS_INDEX_GROUPS.format(table="groups", cluster=CLICKHOUSE_CLUSTER))
+
+
+operations = [
+    migrations.RunPython(add_columns_to_required_tables),
```

Comment:
You should use run_sql_with_exceptions, it executes the queries on all nodes without using `ON CLUSTER`.

Reply:
@orian Thanks! 14255e9a23

Do I need to specify a `NodeRole` too?

Reply:
If a table is sharded then it should be only DATA nodes, otherwise i think it should be run on ALL nodes.

Reply:
3282dfcb022c4a3fa128fee0cfdae1e36279d668